lib/internal/errors.js

```diff
@@ -286,45 +307,9 @@ function makeNodeErrorWithCode(Base, key) {
 
 // This function removes unnecessary frames from Node.js core errors.
 function hideStackFrames(fn) {
-  return function hidden(...args) {
-    // Make sure the most outer `hideStackFrames()` function is used.
-    let setStackFn = false;
-    if (excludedStackFn === undefined) {
-      excludedStackFn = hidden;
-      setStackFn = true;
-    }
-    try {
-      return fn(...args);
-    } finally {
-      if (setStackFn === true) {
-        excludedStackFn = undefined;
-      }
-    }
-  };
-}
-
-function addCodeToName(err, name, code) {
-  // Set the stack
-  if (excludedStackFn !== undefined) {
-    // eslint-disable-next-line no-restricted-syntax
-    Error.captureStackTrace(err, excludedStackFn);
-  }
-  // Add the error code to the name to include it in the stack trace.
-  err.name = `${name} [${code}]`;
-  // Access the stack to generate the error message including the error code
-  // from the name.
-  err.stack;
-  // Reset the name to the actual name.
-  if (name === 'SystemError') {
-    ObjectDefineProperty(err, 'name', {
-      value: name,
-      enumerable: false,
-      writable: true,
-      configurable: true
-    });
-  } else {
-    delete err.name;
-  }
+  const hidden = '__node_internal_hidden_' + fn.name;
+  ObjectDefineProperty(fn, 'name', { value: hidden });
```

Comment:
These names are going to show up during profiling and likely while using a debugger. Changing the function names might therefore confuse people.

Reply:
You still get the original name, but with a prefix (maybe I should reduce the prefix length) and besides, before, you were getting one ```hidden``` call every two frames - I think that if it is not better, it is at least not worse than before

Reply:
My main pain point is that the name does not correspond with the actual function name anymore.

Reply:
Yes, that's the main disadvantage of the approach.

Reply:
```CallSite.getFunction()``` would have allowed for a much cleaner approach. Alas, it is getting removed in the next V8 version. And it was already gimped in strict mode.
<https://www.chromestatus.com/feature/5975936472186880>